src/globals/scss/objects/_form-group.scss

```diff
@@ -0,0 +1,23 @@
+@include exports("form-group") {
+
+  .govuk-o-form-group {
+    @include govuk-responsive-margin($govuk-spacing-responsive-6, "bottom");
+
+    .govuk-o-form-group:last-of-type,
+    .govuk-c-input:last-of-type,
+    .govuk-c-select:last-of-type {
+      margin-bottom: 0; // Remove margin from last item in nested groups
+    }
+  }
+
+  .govuk-o-form-group--error {
+    padding-left: $govuk-spacing-scale-3;
+    border-left: $govuk-border-width-form-group-error solid $govuk-red;
+
+    .govuk-o-form-group {
```

Comment:
Is this the responsibility of the CSS or the use of markup / macros?

Reply:
Yes absolutely, this was an attempt to mitigate the user making a mistake and/or nesting form groups in a way that makes checking for this difficult. In some ways this _is_ more of a visual concern as any form field that has an error class does in fact contain an error, it's just that we shouldn't be showing left error borders within left error borders. I don't know if I'm making any sense...